src/corehost/cli/fxr/fx_muxer.cpp

```diff
@@ -333,67 +366,79 @@ pal::string_t fx_muxer_t::resolve_fx_dir(host_mode_t mode, const pal::string_t&
-            fx_ver_t ver(-1, -1, -1);
-            if (!specified.is_prerelease() && fx_ver_t::parse(version, &ver, true) && // true -- only prod. prevents roll forward to prerelease.
-                ver.get_major() == specified.get_major() &&
-                ver.get_minor() == specified.get_minor())
-            {
-                // Pick the greatest production that differs only in patch.
-                most_compatible = std::max(ver, most_compatible);
-            }
-            if (specified.is_prerelease() && fx_ver_t::parse(version, &ver, false) && // false -- implies both production and prerelease.
-                ver.is_prerelease() && // prevent roll forward to production.
-                ver.get_major() == specified.get_major() &&
-                ver.get_minor() == specified.get_minor() &&
-                ver.get_patch() == specified.get_patch() &&
-                ver > specified)
-            {
-                // Pick the smallest prerelease that is greater than specified.
-                most_compatible = (most_compatible == specified) ? ver : std::min(ver, most_compatible);
-            }
-        }
-        pal::string_t most_compatible_str = most_compatible.as_str();
-        append_path(&fx_dir, most_compatible_str.c_str());
-    }
-
-    trace::verbose(_X("Chose FX version [%s]"), fx_dir.c_str());
-    return fx_dir;
+	pal::string_t env_lookup;
+	std::vector<pal::string_t> hive_dir;
+
+	//Multi-level SharedFX lookup happens only if dotnet_lookup env var is set to 1
+	if (pal::getenv(_X("DOTNET_ORDERED_LOOKUP"), &env_lookup) && env_lookup == _X("1"))
+	{
+		pal::string_t cwd;
+		pal::string_t local_dir;
+		pal::string_t global_dir;
+
+		if (pal::getcwd(&cwd))
+		{
+			hive_dir.push_back(cwd);
+		}
+		if (pal::get_local_dotnet_dir(&local_dir))
+		{
+			hive_dir.push_back(local_dir);
+		}
+		hive_dir.push_back(own_dir);
+		if (pal::get_global_dotnet_dir(&global_dir))
+		{
+			hive_dir.push_back(global_dir);
+		}
+	}
+	else
+	{
+		hive_dir.push_back(own_dir);
+	}
+
+	for (pal::string_t dir : hive_dir)
+	{
+		auto fx_dir = dir;
+		trace::verbose(_X("Searching FX directory in [%s]"), fx_dir.c_str());
+
+		append_path(&fx_dir, _X("shared"));
+		append_path(&fx_dir, fx_name.c_str());
+
+		bool do_roll_forward = false;
+		if (specified_fx_version.empty())
+		{
+			if (!specified.is_prerelease())
+			{
+				// If production and no roll forward use given version.
+				do_roll_forward = config.get_patch_roll_fwd();
+			}
+			else
+			{
+				// Prerelease, but roll forward only if version doesn't exist.
+				pal::string_t ver_dir = fx_dir;
+				append_path(&ver_dir, fx_ver.c_str());
+				do_roll_forward = !pal::directory_exists(ver_dir);
+			}
+		}
+
+		if (!do_roll_forward)
+		{
+			trace::verbose(_X("Did not roll forward because specified version='%s', patch_roll_fwd=%d, chose [%s]"), specified_fx_version.c_str(), config.get_patch_roll_fwd(), fx_ver.c_str());
+			append_path(&fx_dir, fx_ver.c_str());
```

Comment:
It's the second situation.
"Did not roll forward because specified version='', patch_roll_fwd=0, chose [1.0.0]"